eth/downloader/downloader.go

```diff
@@ -793,8 +758,10 @@ func (d *Downloader) findAncestor(p *peerConnection, remoteHeader *types.Header)
 					known = d.blockchain.HasBlock(h, n)
 				case FastSync:
 					known = d.blockchain.HasFastBlock(h, n)
-				default:
+				case LightSync:
 					known = d.lightchain.HasHeader(h, n)
+				default:
+					panic("unknown sync mode: " + d.mode.String())
```

Comment:
This is a new potential but theoretically-noop logic change, and like all audacious panics, it thinks it never should or will happen. The `d.mode.String()` will return "unknown."